core/src/main/scala/kafka/cluster/Replica.scala

```diff
@@ -35,6 +34,9 @@ class Replica(val brokerId: Int,
   // the log end offset value, kept in all replicas;
   // for local replica it is the log's end offset, for remote replicas its value is only updated by follower fetch
   @volatile private[this] var logEndOffsetMetadata = LogOffsetMetadata.UnknownOffsetMetadata
+  // the log start offset value, kept in all replicas;
+  // for local replica it is the log's start offset, for remote replicas its value is only updated by follower fetch
+  @volatile private[this] var _logStartOffset = 0L
```

Comment:
There is no existing variable named UnknownOffset. There is `LogOffsetMetadata.UnknownSegBaseOffset` but I am not sure we should use UnknownSegBaseOffset for logStartOffset. So I created `val UnknownLogStartOffset = -1L`. Does this work?